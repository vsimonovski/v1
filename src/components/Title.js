import React from 'react';
import './Title.scss';

export default function Title() {
  return (
    <section>
      <h1 className="intro intro--wave">
        Hi, I'm Vladimir
        <span className="background-line"></span>
      </h1>
      <br />
      <h1 className="intro intro--pulse">
        I
        <span className="emoji" role="img" aria-label="heart">
          ❤️
        </span>
        Web
        <span className="background-line"></span>
      </h1>
    </section>
  );
}
