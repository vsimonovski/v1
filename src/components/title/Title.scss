$color__blue: #0068fa;

.intro {
  display: inline-block;
  margin: 0;
  position: relative;
  z-index: 1;
  font-family: 'ChunkFive', serif;
  font-weight: 700;
  font-size: 7.2rem;
  letter-spacing: 1.5px;

  &--pulse {
    position: relative;
  }

  &--wave {
    margin-left: -73px;
    &::before {
      content: 'ðŸ‘‹';
      animation-name: wobble;
      animation-duration: 1s;
      animation-delay: 1s;

      animation-fill-mode: both;
      font-size: 5.4rem;
      display: inline-block;
      padding-top: 20px;
      margin-right: 18px;
    }

    .background-line {
      &::before {
        left: 72px;
      }
    }
  }
}

.emoji {
  padding: 10px;
  font-size: 5.4rem;
}

.background-line {
  &::before {
    background-color: $color__blue;
    position: absolute;
    top: 45%;
    left: 0px;
    right: 5px;
    height: 13px;
    margin-top: -5px;
    content: '';
    z-index: -1;
  }
}
