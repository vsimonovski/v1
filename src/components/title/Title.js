import React from 'react';
import { BackgroundLine, EmojiWithAttrs, Intro, Wave } from './Title.style';

export default function Title() {
  return (
    <section>
      <Intro>
        <Wave />
        Hi, I'm Vladimir
        <BackgroundLine />
      </Intro>
      <br />
      <Intro>
        {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}I
        <EmojiWithAttrs>❤️</EmojiWithAttrs>
        Web
        <BackgroundLine />
      </Intro>
    </section>
  );
}
