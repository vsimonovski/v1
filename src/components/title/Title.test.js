import React from 'react';
import { render } from '@testing-library/react';
import Title from './Title';

it('should match snapshot', () => {
  const tree = render(<Title />);
  expect(tree).toMatchSnapshot();
});
