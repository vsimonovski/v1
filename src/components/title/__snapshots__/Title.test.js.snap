// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h1
          class="sc-hKFyIo eKOBkk"
        >
          <span
            class="sc-eCApGN jPjvkh"
          />
          Hi, I'm Vladimir
          <span
            class="sc-bdnylx direLq"
          />
        </h1>
        <br />
        <h1
          class="sc-hKFyIo eKOBkk"
        >
          I
          <span
            aria-label="heart"
            class="sc-gtssRu sc-dlnjPT fEqEBU"
            role="img"
          >
            ❤️
          </span>
          Web
          <span
            class="sc-bdnylx direLq"
          />
        </h1>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <h1
        class="sc-hKFyIo eKOBkk"
      >
        <span
          class="sc-eCApGN jPjvkh"
        />
        Hi, I'm Vladimir
        <span
          class="sc-bdnylx direLq"
        />
      </h1>
      <br />
      <h1
        class="sc-hKFyIo eKOBkk"
      >
        I
        <span
          aria-label="heart"
          class="sc-gtssRu sc-dlnjPT fEqEBU"
          role="img"
        >
          ❤️
        </span>
        Web
        <span
          class="sc-bdnylx direLq"
        />
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
